lxd/storage/drivers/clients/alletra_wsapi1.go

```diff
@@ -667,3 +675,94 @@ func (p *AlletraClient) DeleteVolume(poolName string, volName string) error {
 
 	return p.deleteVolume(poolName, volName)
 }
+
+// ConnectHostToVolume creates a connection between a host and volume. It returns true if the connection
+// was created, and false if it already existed.
+func (p *AlletraClient) ConnectHostToVolume(poolName string, volName string, hostName string) (bool, error) {
+	url := api.NewURL().Path("api", "v1", "vluns")
+
+	req := make(map[string]any)
+
+	req["hostname"] = hostName
+	req["volumeName"] = volName
+	req["lun"] = 0
+	req["autoLun"] = true
+
+	err := p.requestAuthenticated(http.MethodPost, url.URL, req, nil)
+	if err != nil {
+		hpeErr, ok := err.(*hpeError)
+		if ok {
+			switch hpeErr.Code {
+			case apiErrorVolumeIsAlreadyExported:
+				logger.Debugf("HPE using existing settings. Volume %s already attached to %s", volName, hostName)
+				return false, nil
+			default:
+				return false, fmt.Errorf("HPE debug. hpeErr.Code: %d. hpeErr.Desc: %s", hpeErr.Code, hpeErr.Desc)
+			}
+		}
+
+		return false, fmt.Errorf("Failed to connect volume %q with host %q: %w", volName, hostName, err)
+	}
+
+	return true, nil
+}
+
+// DisconnectHostFromVolume deletes a connection between a host and volume.
+func (p *AlletraClient) DisconnectHostFromVolume(poolName string, volName string, hostName string) error {
+	vlun, errVLUN := p.GetVLUN(volName)
+	if errVLUN != nil {
+		return fmt.Errorf("HPE Error %w", errVLUN)
+	}
+
+	if vlun == nil {
+		logger.Debugf("HPE debug no need to disconnect host %s from volume %s", hostName, volName)
+		return nil
+	}
+
+	customParam := volName + "," + strconv.Itoa(vlun.LUN) + "," + hostName
+	url := api.NewURL().Path("api", "v1", "vluns", customParam)
+
+	err := p.requestAuthenticated(http.MethodDelete, url.URL, nil, nil)
+	if err != nil {
+		if isHpeErrorNotFound(err) {
+			return api.StatusErrorf(http.StatusNotFound, "Connection between host %q and volume %q not found", volName, hostName)
+		}
+
+		return fmt.Errorf("Failed to disconnect volume %q from host %q: %w", volName, hostName, err)
+	}
+
+	return nil
+}
+
+// GetVLUN returns VLUN related data of given volumeName.
+func (p *AlletraClient) GetVLUN(volumeName string) (*hpeVLUN, error) {
+	var resp hpeRespMembers[hpeVLUN]
+
+	url := api.NewURL().Path("api", "v1", "vluns").WithQuery("query", "\"volumeName"+"=="+volumeName+"\"")
```

Comment:
the query string here caught my eye, do we need to quote the `query` value?
If so can https://pkg.go.dev/strconv#Quote do the job?